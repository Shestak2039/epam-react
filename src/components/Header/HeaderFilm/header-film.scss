@import "../../../styles/colors";

.header-top {
  display: flex;
  justify-content: flex-end;

  margin: 0 30px;
}

.header-background {
  height: 100%;
  width: 100%;
  background-color: #000000BB;

  padding-top: 15px;
  padding-bottom: 15px;

  color: $default-text-color;
}

.header-film {
  display: flex;

  margin: 20px 90px;

  .header-film-image {
    margin-right: 40px;
  }

  .header-film-info-title-and-rating {
    display: flex;

    .header-film-info-title {
      font-size: 36px;

      margin-right: 10px;
    }

    .header-film-info-rating {
      display: flex;
      justify-content: center;
      align-items: center;

      width: 35px;
      height: 35px;

      padding: 5px;

      border: 1px solid $default-text-color;
      border-radius: 61%;

      font-size: 20px;
      color: $rating-text-color;
    }
  }

  .header-film-info-genres {
    color: $film-details-color;
  }

  .header-film-info-year-and-duration {
    display: flex;

    margin: 15px 0;

    font-size: 24px;
    color: $pink-text-color;

    .header-film-info-year {
      margin-right: 25px;
    }
  }

  .header-film-info-overview {
    color: $film-details-color;
  }
}
