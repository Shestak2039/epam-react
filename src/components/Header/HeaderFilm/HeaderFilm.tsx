import React from 'react';

import './header-film.scss';
import { FilmCardModel } from '../../FilmsList/FilmCard/FilmCard';
import Button from '../../common/Button/Button';

export interface HeaderFilmProps {
    film: FilmCardModel;
    openPage: (id: number | undefined) => void;
}

const HeaderFilm: React.FunctionComponent<HeaderFilmProps> = (
    {film: {title, imageUrl, rating, genres, year, durationInMinutes, overview}, openPage}) => {
    return (
        <div className="header-background">
            <div className="header-top">
                <Button title={'Search'} primary={false} action={() => openPage(undefined)}/>
            </div>
            <div className="header-film">
                <img className="header-film-image" src={imageUrl} height="350" width="250" />
                <div className="header-film-info">
                    <div className="header-film-info-title-and-rating">
                        <div className="header-film-info-title">{title}</div>
                        <div className="header-film-info-rating">{rating}</div>
                    </div>
                    <div className="header-film-info-genres">{genres.join(', ')}</div>
                    <div className="header-film-info-year-and-duration">
                        <div className="header-film-info-year">{year}</div>
                        <div className="header-film-info-duration">{durationInMinutes + 'min'}</div>
                    </div>
                    <div className="header-film-info-overview">{overview}</div>
                </div>
            </div>
        </div>
    );
};

export default HeaderFilm;
