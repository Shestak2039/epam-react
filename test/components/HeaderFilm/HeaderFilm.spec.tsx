import React from 'react';
import { render } from '@testing-library/react';

import HeaderFilm from '../../../src/components/Header/HeaderFilm/HeaderFilm';
import { BrowserRouter } from 'react-router-dom';

describe('HeaderFilm', () => {
    test('renders HeaderFilm snapshot', () => {
        const { asFragment } = render(<BrowserRouter><HeaderFilm films={[]}/></BrowserRouter>);

        expect(asFragment()).toMatchSnapshot();
    });
});
