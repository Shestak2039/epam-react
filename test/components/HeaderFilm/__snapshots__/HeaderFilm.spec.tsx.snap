// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderFilm renders HeaderFilm snapshot 1`] = `
<DocumentFragment>
  <div
    class="header-background"
  >
    <div
      class="header-top"
    >
      <a
        href="/home"
      >
        <button
          class="button default"
          data-testid="Search"
        >
          Search
        </button>
      </a>
    </div>
    <div
      class="header-film"
    >
      <img
        class="header-film-image"
        height="350"
        src=""
        width="250"
      />
      <div
        class="header-film-info"
      >
        <div
          class="header-film-info-title-and-rating"
        >
          <div
            class="header-film-info-title"
          />
          <div
            class="header-film-info-rating"
          >
            0
          </div>
        </div>
        <div
          class="header-film-info-genres"
        />
        <div
          class="header-film-info-year-and-duration"
        >
          <div
            class="header-film-info-year"
          />
          <div
            class="header-film-info-duration"
          >
            0min
          </div>
        </div>
        <div
          class="header-film-info-overview"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
